server/src/main/java/org/elasticsearch/ingest/IngestStats.java

```diff
@@ -170,7 +186,20 @@ public static IngestStats merge(IngestStats first, IngestStats second) {
         );
     }
 
-    static Map<String, List<ProcessorStat>> merge(Map<String, List<ProcessorStat>> first, Map<String, List<ProcessorStat>> second) {
+    static Map<ProjectId, Map<String, List<ProcessorStat>>> merge(
```

Comment:
I feel like there's a significantly more efficient way to do this merging in the way it's currently being used... But optimizations are not the focus of the MP work right now and I'm not even sure if any optimizations here are worth it for how often they'll be used.